app/controllers/stories_controller.rb

```diff
@@ -131,6 +136,8 @@ def handle_base_index
     assign_podcasts
     assign_classified_listings
     @article_index = true
+    @featured_story = (@featured_story || Article.new)&.decorate
```

Comment:
I'm not following this part. The featured story is either the featured story or a new Article? Is this new article, just an arbitrary new article?

Reply:
Nope: it's an empty article object. This was originally done at app/views/articles/index.html.erb:87, I suspect to prevent the view template from having to make a ton of nil object checks

Reply:
Silly question maybe, but does this mean if we do not set a `main_image` for an article, we could potentially be displaying an empty article object?

Reply:
Probably not: at `app/controllers/stories_controller.rb:26` we are specifically grabbing the first "hot" article with an image. If there isn't one, we'll set `@featured_story` to an empty `Article` object. At `app/views/articles/index.html.erb:86` there's a check for `if @feature_story.id`. With a brand new `Article` object, it hasn't been saved yet, thus it doesn't have an id and will fail a check (this is a fairly common Ruby pattern of checking if an object has been saved).

So: if we can't find a "hot" article with an image set, we'll end up skipping rendering of the 'featured' story block.

Reply:
This would be a great conversation to summarize in a comment inside the code, as the feed is still somewhat obscure :D